File: InterpreterC/InterpreterC/model/myStack/MyIStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InterpreterC.model.myStack
{
    public interface MyIStack<T>
    {
        void push(T elToPush);

        T pop();

        T peek();

        bool isEmpty();
    }
}

File: JavaToyLanguageInterpreter/src/model/MyIStack.java
package model;
import exceptions.ExceptionMyStack;

public interface MyIStack<T> {
	public T pop() throws ExceptionMyStack;
	public void push(T val);
	public T peek() throws ExceptionMyStack;
	public String toString();
	public boolean is_empty();
	public Object[] toArray();
}
